### PR TITLE
issue-519

### PR DESCRIPTION
**Pull request by robrobert99**

### CODE REVIEW

File: src/main/java/de/bonndan/nivio/assessment/AssessmentFactory.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * A factory to create Assessment instances from map or Landscape
+ */
+
+public class AssessmentFactory {
+
+    public static Assessment createAssessment(Map<FullyQualifiedIdentifier, List<StatusValue>> results) {
+        return new Assessment(results);
+    }
+
+    public static Assessment createAssessment(Landscape landscape) {
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/AssessmentFactory.java:19]: We might think about changing the param to `Map<String, KPI> kpis`, so any combination of assessments can be used.

File: src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Stores and loads and creates Assessments with FullyQualifiedIdentifier as key
+ */
+@Component
+public class AssessmentRepository {
+
+    private HashMap<FullyQualifiedIdentifier, Assessment> repository;
+
+    public AssessmentRepository() {
+        repository = new HashMap<>();
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:19]: Change the repository variable to be final and find a way to clean the state ;)

File: src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        if (landscape != null) {
+            var newAssessment = AssessmentFactory.createAssessment(landscape);
+            storeAssessment(landscape.getFullyQualifiedIdentifier(), newAssessment);
+            return newAssessment;
+        } else {
+            return null;
+        }
+    }
+
+    public Assessment getAssessment(@NonNull FullyQualifiedIdentifier fullyQualifiedIdentifier) {
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:36]: If you use a @NonNull annotation you should ensure it is not null (e.g. use Objects.requireNonNull(...)) or instead annotate @Nullable or nothing. Declaring the parameter as final ensures the "pureness" of the method a bit.

The holy inquisition @Matthimatiker and @basseur might see that more strictly, though.  ✝️

Comment (bonndan) [reply src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:36]: 😉

File: src/main/java/de/bonndan/nivio/search/SearchIndexingEventListener.java [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
+    }
+
     @EventListener(ProcessingFinishedEvent.class)
     public void onProcessingFinishedEvent(@NonNull ProcessingFinishedEvent event) {
         final Landscape landscape = event.getLandscape();
         final SearchIndex searchIndex = landscape.getSearchIndex();
 
         //see https://github.com/dedica-team/nivio/issues/519
-        Assessment assessment = new Assessment(landscape.applyKPIs(landscape.getKpis()));
+        var assessment = assessmentRepository.getAssessment(landscape.getFullyQualifiedIdentifier());
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/search/SearchIndexingEventListener.java:28]: I am not sure if the landscape is already updated in its repo at this time. Please check and lets talk about what is a good approach. Maybe it's all fine.

File: src/test/java/de/bonndan/nivio/assessment/AssessmentRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
+package de.bonndan.nivio.assessment;
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/assessment/AssessmentRepositoryTest.java:1]: 👍

File: src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java (hunk 3 above)
Comment (Matthimatiker) [reply src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:36]: [NOBODY EXPECTS THE INQUISITION](https://imgflip.com/i/3b32u8)

Comment (bonndan) [reply src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:36]: 🤣

File: src/test/java/de/bonndan/nivio/output/docs/DocsControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
+package de.bonndan.nivio.output.docs;
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/output/docs/DocsControllerTest.java:1]: ❤️

File: src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java (hunk 2 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:21]: nice. you can initialize the member right in line 18, but this does it, too.

File: src/main/java/de/bonndan/nivio/assessment/AssessmentFactory.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A factory to create Assessment instances from map or Landscape
+ */
+
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/AssessmentFactory.java:16]:
```suggestion
```

File: src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    public AssessmentRepository() {
+        repository = new HashMap<>();
+    }
+
+    public void clean() {
+        repository.clear();
+    }
+
+    public Assessment createAssessment(@NonNull Landscape landscape) {
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:28]:
```suggestion
   @NonNull
    public Assessment createAssessment(@NonNull Landscape landscape) {
```

Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java:28]: Public methods, especially in a widely used factory should have javadoc explaining what is happening.

**Pull request merged**